src/drivers/dw/dma.c

```diff
@@ -190,6 +190,12 @@ static struct dma_chan_data *dw_dma_channel_get(struct dma *dma,
 	tr_info(&dwdma_tr, "dw_dma_channel_get(): dma %d request channel %d",
 		dma->plat_data.id, req_chan);
 
+	dcache_invalidate_region(dma,
+				 sizeof(struct dma));
+
+	dcache_invalidate_region(dma->chan,
+				 sizeof(struct dma_chan_data) * dma->plat_data.channels);
+
```

Comment:
These should be after the spin_lock (since the lock has a memory barrier built in)